README.md

```diff
@@ -12,60 +13,70 @@ the table of current chrome and the associated v8 releases at:
 
   http://omahaproxy.appspot.com/
 
-# Compiling
-
-In order for the bindings to compile correctly, one needs to:
-
-1. Compile v8 as a static library.
-2. Let cgo know where the library is located.
-
-## Compiling v8.
-
-Download v8: https://github.com/v8/v8/wiki/Using%20Git
-
-Lets say you've checked out the v8 source into `$V8`, go-v8 into `$GO_V8` and
-want to place the static v8 library into `$GO_V8/libv8/`.  (For example,
-`export GO_V8=$GOPATH/src/github.com/augustoroman/v8`)
-
-
-### Linux
-
-Build:
-
-    make native GYPFLAGS="-Dv8_use_external_startup_data=0 -Dv8_enable_i18n_support=0 -Dv8_enable_gdbjit=0"
-
-If build system produces a thin archive, you want to make it into a fat one:
-
-    for lib in `find out/native/obj.target/src/ -name '*.a'`;
-      do ar -t $lib | xargs ar rvs $lib.new && mv -v $lib.new $lib;
-    done
-
-Symlink the libraries and include directory to the Go package dir:
-
-    ln -s `pwd`/out/native/obj.target/src ${GO_V8}/libv8
-    ln -s `pwd`/include ${GO_V8}/include
-
-
-### Mac
-
-To build: (substitute in your OS X version: `sw_vers -productVersion`)
-
-    GYP_DEFINES="mac_deployment_target=10.11" \
-    make -j5 native GYPFLAGS="-Dv8_use_external_startup_data=0 -Dv8_enable_i18n_support=0 -Dv8_enable_gdbjit=0"
+# Building  v8
+## Prep
+You need to build v8 statically and place it in a location cgo knows about. This requires special tooling and a build directory. Using the [official instructions](https://github.com/v8/v8/wiki/Building-from-Source) as a guide, the general steps of this process are:
+
+1. `go get` the binding library (this library)
+1. Create a v8 build directory
+1. [Install depot tools](http://commondatastorage.googleapis.com/chrome-infra-docs/flat/depot_tools/docs/html/depot_tools_tutorial.html#_setting_up)
+1. Configure environment
+1. Download v8
+1. Build v8
+1. Copy or symlink files to the go library path
+1. Build the bindings
+
+```
+go get github.com/augustoroman/v8
+export V8_GO=$GOPATH/src/github.com/augustroman/v8
+export V8_BUILD=$V8_GO/v8build #or wherever you like
+mkdir -p $V8_BUILD
+cd $V8_BUILD
+git clone https://chromium.googlesource.com/chromium/tools/depot_tools.git
+export PATH=$PATH:$V8_BUILD/depot_tools
+fetch v8 #pull down v8 (this will take some time)
+cd v8
+gclient sync
+```
+
+## Linux
+```
+./build/install-build-deps.sh #only needed once
+gn gen out.gn/golib --args="is_official_build=true strip_debug_info=true v8_use_external_startup_data=false v8_enable_i18n_support=false v8_enable_gdbjit=false v8_static_library=true"
+ninja -C out.gn/golib
+# go get some coffee
+```
+
+## OSX
+
+```
+export V8_OSX_SDK_VERSION=`xcrun --sdk macosx --show-sdk-path|awk -F "/" '{print $NF}'|grep -o '\d\+\.\d\+'`
+gn gen out.gn/golib --args="mac_deployment_target=\"$V8_OSX_SDK_VERSION\" is_official_build=true strip_debug_info=true v8_use_external_startup_data=false v8_enable_i18n_support=false v8_enable_gdbjit=false v8_static_library=true"
```

Comment:
two things:
1: do you need `mac_deployment_target`?
2: why do you include `is_official_build`?

Reply:
# `mac_deployment_target`
Hmm, good question.  All I can tell you so far is that using your instructions I get erros like the above.  If I don't specify it, it looks like it will default to `10.9.0` as per `gn args out.gn/golib --list`:

```
mac_deployment_target
    Current value = "10.13"
      From //out.gn/golib/args.gn:1
    Overridden from the default = "10.9.0"
      From //build/config/mac/mac_sdk.gni:12

    Minimum supported version of macOS. Must be of the form x.x.x for
    Info.plist files.
```

#`is_official_build`
Well, according to `gn`, it's supposed to be the following:
```
is_official_build
    Current value = true
      From //out.gn/golib/args.gn:2
    Overridden from the default = false
      From //build/config/BUILDCONFIG.gn:131

    Set to enable the official build level of optimization. This has nothing
    to do with branding, but enables an additional level of optimization above
    release (!is_debug). This might be better expressed as a tri-state
    (debug, release, official) but for historical reasons there are two
    separate flags.
```

It's unclear from these docs whether this sets `is_debug = false` or not, but it seems to forget to remove the debugging symbols either way, so ¯\_(ツ)_/¯. They should probably both be included "for historical reasons" anyway.

Reply:
Thanks for digging into this.  There are a couple more flags that I put in while trying to get this working reliably under travis-ci that I found online that may or may not help:

```
    symbol_level = 0
    v8_experimental_extra_library_files = []
    v8_extra_library_files = []
```
I'm really not confident whether the v8_* flags are useful, but the symbol_level = 0 appears to also try to remove debug symbols.  If it doesn't provide any improvement over is_official_build, then we'll go with that since it's simpler.

Reply:
`v8_static_library` definitely seems necessary, as many of the .a files didn't exist when I built without it.